AppDashboard/lib/app_dashboard_data.py

```diff
@@ -647,3 +729,125 @@ def can_upload_apps(self):
     except Exception as err:
       logging.exception(err)
       return False
+
+  def set_dash_layout_settings(self, values=None):
+    """ Saves user settings
+        for customizing the UI of the Dashboard.
+
+    Args:
+      values: A JSON str that defines the layout of the dash page
+        from /ajax/save/layout
+    """
+    user = users.get_current_user()
+    if not user:
+      return
+    email = user.email()
+    try:
+      user_info = self.get_by_id(UserInfo, email)
+      if user_info:
+        try:
+          temp_dict = json.loads(values)
+        except TypeError as err:
```

Comment:
In addition to catching invalid JSON, you should also verify that there are no unexpected keys or values. If a user without appropriate permissions posted values that only a cloud admin should be able to see, it probably shouldn't be written to the database even though you filter it when retrieving that data.

It would make sense to me to have the validation happen in the post handler, but it could also go here instead.

As a side note, I think it would make more sense for this function to raise an exception on invalid input rather than continue on with default values as if nothing went wrong.